src/configure/services/source/sourceRepositoryService.ts

```diff
@@ -0,0 +1,126 @@
+import { AzureDevOpsService } from "../azureDevOpsService";
+import Q = require('q');
+import { GitHubProvider } from '../gitHubService';
+import { BranchSummary } from 'simple-git/typings/response';
+
+export class SourceRepositoryService {
+    private gitReference: git.SimpleGit;
+
+    public async getGitRepoDetails(repositoryPath: string): Promise<GitRepositoryDetails> {
+        let gitPath: string = path.join(repositoryPath, '.git');
+        if (!fs.existsSync(gitPath)) {
+            throw new Error(`Git folder could not be found inside the folder at path: ${repositoryPath}`);
+        }
+
+        this.gitReference = git(repositoryPath);
+        let status = await this.gitReference.status();
+        let branch = status.current;
+        let commitId = await this.getLatestCommitId(branch);
+        let tracking = status.tracking.substr(0, status.tracking.indexOf(branch) - 1);
+        let remoteUrl = await this.gitReference.remote(["get-url", tracking]);
+
+        if (remoteUrl) {
+            if (AzureDevOpsService.isAzureReposUrl(remoteUrl)) {
+                return <GitRepositoryDetails> {
+                    sourceProvider: SourceProviderType.AzureRepos,
+                    repositoryId: "",
+                    repositoryName: AzureDevOpsService.getRepositoryNameFromRemoteUrl(remoteUrl),
+                    remoteUrl: remoteUrl,
+                    branch: branch,
+                    commitId: commitId,
+                    localPath: repositoryPath
+                };
+            }
+            else if (GitHubProvider.isGitHubUrl(remoteUrl)) {
+                let repoId = GitHubProvider.getRepositoryIdFromUrl(remoteUrl);
+                return <GitRepositoryDetails>{
+                    sourceProvider: SourceProviderType.Github,
+                    repositoryId: repoId,
+                    repositoryName: repoId,
+                    remoteUrl: remoteUrl,
+                    branch: branch,
+                    commitId: commitId,
+                    localPath: repositoryPath
+                };
+            }
+            else {
+                throw new Error("Could not identify repository details. Ensure your git repo is managed with Azure Repos or Github");
+            }
+        }
+        else {
+            throw new Error("Remote repository is not configured. Manage your git repository with Azure Repos or Github");
+        }
+    }
+
+    /**
+     *
+     * @param pipelineYamlPath : local path of yaml pipeline in the extension
+     * @param context: inputs required to be filled in the yaml pipelines
+     * @returns: thenable object which resolves once all files are added to the repository
+     */
+    public addYmlFileToRepo(ymlFilePath: string, repoPath: string, context: any): Q.Promise<string> {
+        let deferred: Q.Deferred<string> = Q.defer();
+        fs.readFile(ymlFilePath, { encoding: "utf8" }, async (error, data) => {
```

Comment:
How does this work if the file is not present

Reply:
If the file already exists - we should create one more

Reply:
ymlFilePath is the path to pipelineTemplate in our extension.

Reply:
Now creating a new file if file with that name already exists.